dashboard/app/models/user.rb

```diff
@@ -1759,6 +1759,8 @@ def show_school_info_interstitial?
   def clear_user_and_mark_purged
     random_suffix = (('0'..'9').to_a + ('a'..'z').to_a).sample(8).join
 
+    authentication_options.with_deleted.each(&:really_destroy!)
```

Comment:
Do you remember the reason we're not in general calling user's `really_destroy!`? I'm just wondering if the same logic would suggest that we purge `authentication_options` as well?

Reply:
Primarily for metrics purposes.  On a user data purge request we are fully anonymizing the user row instead of purging it so that we can still pull aggregate stats about levels completed by our population, etc.  Apparently this is compliant :man_shrugging:.  See [spec 1](https://docs.google.com/document/d/1WD7OQXyJMiAP2x-ct7mc29pU0daLpHVaaWXTdkB1mtQ/edit) and [spec 2](https://docs.google.com/spreadsheets/d/1RybuZunt5F6zSXp33oC4KdaBl20VMHVi0yprI8P30y0/edit#gid=670705846) for details.

I'm using `really_destroy!` on authentication_options because the spec says so, which in turn is probably because they are not very relevant to our KPIs and very likely to be almost entirely made up of PPII, which we do need to remove.